### PR TITLE
update calc_creat for adults with NHANES regression

### PR DESCRIPTION
**Pull request by JordanBrooks33**
Added my regression model to calc_creat() and updated tests. Would love recommendations on how to handle warnings for height and weight as these were historically not required by this function. Since they were weaker effects than Age and Sex, I think imputing the median is fine in cases where they are not available. Happy to alter.

### CODE REVIEW

**Review by jasmineirx — COMMENTED** [review]
I'm excited to add this new functionality! 

I think best practices would be to add a new argument for `method`, i.e.,

```r
method = c("Johanssen", "NHANES"),
```

then one the the early lines in the function can be
```
method <- match.arg(method)
```
which means if no method is supplied, the old logic is used, and only accepted methods are allowed.

we can then change the warnings accordingly:

```r
if (is.null(weight) && method == "NHANES") stop("Weight is required)
```
etc

The body of the function would also have to be updated to that both the old and the new logic is used depending on method. We might want to consider putting it in functions and calling the right calculation method based on a `switch` statement. see `calc_ffm` for an example.

I also suggest using the variable names `weight` and `height`, since that is consistent with what we use elsewhere in the package

File: R/calc_creat.R [hunk 1]
```diff
@@ -4,19 +4,27 @@
 #'
 #' Uses equations described in Ceriotti et al. Clin Chem. 2008, and Junge W et al.
 #' Clin Chim Acta. 2004. For age 15-18, a linear interpolation is used between
-#' equations for <15 and >18 years as described in Johanssen A et al. Ther Drug
-#' Monit 2011.
+#' equations for <15 as described in Johanssen A et al. Ther Drug
+#' Monit 2011. Predictions for age >18 are based on an linear regression 
+#' built on open-source data (NHANES). Median height and weight imputed if not 
+#' specified.
```
Comment (roninsightrx) [inline R/calc_creat.R:10]: Can we put the core of the linear regression fit code that was used here in the description, for reproducibility? I guess was sth like

```
lm(creat ~ weight + height + sex, data = NHANES)
```
Or if you split it up in separate functions like Jasmine suggested then add it there.

File: R/calc_creat.R [hunk 2]
```diff
@@ -25,14 +33,21 @@ calc_creat <- function (
...
+    warning("165 cm used: recommend specifying ht (cm)")
+  }
+    scr <- rep(0, length(age))
+    scr[age <= 15] <- -2.37330 -(12.91367 * log(age[age <= 15])) + 23.93581 * sqrt(age[age <= 15])
+    sel1 <- (age > 15 & age < 18) & sex == "male"
+    scr[sel1]   <- 9.5471 * age[sel1] - 87.847
+    sel2 <- (age > 15 & age < 18) & sex == "female"
+    scr[sel2] <- 4.7137 * age[sel2] - 15.347
+    weight <-  ifelse(is.null(wt),75,wt)
+    height <-ifelse(is.null(ht),165,ht)
```
Comment (roninsightrx) [inline R/calc_creat.R:49]: spacing
```suggestion
    weight <- ifelse(is.null(wt), 75, wt)
    height <- ifelse(is.null(ht), 165, ht)
```

File: tests/testthat/test_calc_creat.R [hunk 3]
```diff
@@ -1,10 +1,51 @@
 test_that("calc_create calculates creatinine estimates", {
   expect_equal(
-    calc_creat(
+    suppressWarnings(calc_creat(
```
Comment (roninsightrx) [inline tests/testthat/test_calc_creat.R:3]: if we expect and intended to have a warning, we should use `expect_warning()`. That will also suppress it, but test for it at the same time.

**Review by roninsightrx — CHANGES_REQUESTED** [review]
agree with Jasmine's comments. Few minor add'l things

**Comment by JordanBrooks33** [comment]
Okay this all makes sense. I have altered the logic to include an adult_method specification.  In this way children and adolescents are handled the same way; however, it will default to the historic johanssen flat scr method for adults.

File: R/calc_creat.R [hunk 4]
```diff
@@ -2,54 +2,93 @@
...
+  height = NULL,
+  adult_method = c("johanssen", "brooks"),
+  digits = 1 ) {
+  adult_method <- match.arg(adult_method)
+  scr <- switch(
+    adult_method,
+    "johanssen" = calc_creat_johanssen(
+      age = age,
+      sex = sex,
+      adult_method=adult_method
```
Comment (jasmineirx) [inline R/calc_creat.R:39]:
```suggestion
      adult_method = adult_method
```

File: R/calc_creat.R [hunk 5]
```diff
@@ -2,54 +2,93 @@
...
+    ),
+    "brooks" = calc_creat_brooks(
+      age = age,
+      sex = sex,
+      weight = weight,
+      height = height,
+      adult_method=adult_method
```
Comment (jasmineirx) [inline R/calc_creat.R:46]:
```suggestion
      adult_method = adult_method
```

File: R/calc_creat.R [hunk 6]
```diff
@@ -2,54 +2,93 @@
...
+    )
+  )
+  return(list(
+    value = round(scr, digits),
+    unit = "micromol/L",
+    method = tolower(adult_method)
+  ))
+  calc_creat_johanssen <- function(age, sex, adult_method) {
```
Comment (jasmineirx) [inline R/calc_creat.R:54]:
```suggestion
  ))
  
  calc_creat_johanssen <- function(age, sex, adult_method) {
```

File: R/calc_creat.R [hunk 7]
```diff
@@ -2,54 +2,93 @@
...
+    if(is.null(sex) || !all(sex %in% c("male", "female"))) {
+      stop("Sex needs to be either male or female!")
+    }
+    if(is.null(age)) {
+      stop("Age required.")
+    }
+    scr <- rep(0, length(age))
+    scr[age <= 15] <- -2.37330 -(12.91367 * log(age[age <= 15])) + 23.93581 * sqrt(age[age <= 15])
+    sel1 <- (age > 15 & age < 18) & sex == "male"
+    scr[sel1]   <- 9.5471 * age[sel1] - 87.847
```
Comment (jasmineirx) [inline R/calc_creat.R:64]:
```suggestion
    scr[sel1] <- 9.5471 * age[sel1] - 87.847
```

File: tests/testthat/test_calc_creat.R [hunk 8]
```diff
@@ -1,64 +1,55 @@
...
+      sex = c("male", "male"),
+      age = c(20,80),
+      weight = c(50,120),
+      height = c(120,200),
+      adult_method="brooks")$value,
+    c(55.5, 109.4)
+  )
+}
+
+
```
Comment (jasmineirx) [inline tests/testthat/test_calc_creat.R:55]: we should also test for appropriate error handling if age/sex/weight/height not supplied.

File: R/calc_creat.R [hunk 9]
```diff
@@ -2,54 +2,93 @@
...
+    sel2 <- (age > 15 & age < 18) & sex == "female"
+    scr[sel2] <- 4.7137 * age[sel2] - 15.347
+    scr[age >= 18 & sex == "male"]   <- 84
+    scr[age >= 18 & sex == "female"] <- 69.5
+    scr
+  }
+  calc_creat_brooks <- function(age, sex, weight, height, adult_method) {
```
Comment (jasmineirx) [inline R/calc_creat.R:71]:
```suggestion
  }
  
  calc_creat_brooks <- function(age, sex, weight, height, adult_method) {
```

File: R/calc_creat.R [hunk 10]
```diff
@@ -2,54 +2,93 @@
...
+      sex = sex,
+      weight = weight,
+      height = height,
+      adult_method=adult_method
+    )
+  )
+  return(list(
+    value = round(scr, digits),
+    unit = "micromol/L",
+    method = tolower(adult_method)
```
Comment (jasmineirx) [inline R/calc_creat.R:52]: returning `method` is a new behavior and we should add a test for this too (and for unit)

**Review by jasmineirx — CHANGES_REQUESTED** [review]
Thanks for refactoring. This looks great, and it's very readable. I have just a couple white spacing requests to bring this in line with the [tidyverse style guide](https://style.tidyverse.org/). There's also a couple behaviors we are not currently testing that we should. Please see the `expect_error` function.

**Review by jasmineirx — CHANGES_REQUESTED** [review]
something odd seems to have happened in this commit, where we are now defining the helper functions within the scope of the main function. We should instead define these functions outside of the function (as they were before)

File: R/calc_creat.R [hunk 11]
```diff
@@ -2,39 +2,98 @@
...
-#' equations for <15 and >18 years as described in Johanssen A et al. Ther Drug
-#' Monit 2011.
+#' For children and adolescents: Uses equations described in Ceriotti et al. 
+#' Clin Chem. 2008, and Junge W et al. Clin Chim Acta. 2004. For age 15-18, 
+#' a linear interpolation is used between equations for <15 Johanssen A et al. 
+#' Ther Drug Monit 2011.
+#' For adults:  Two methods are available Johanssen et al. Ther Drug Monit 2011.
+#' which describes a flat serum creatinine dependent on sex, and Brooks et al 
+#' (unpublished) which is a linear regression built on the NHANES open-source
+#' data and takes into account sex, age, weight, and height. Equation: lm(SCr~Age+Sex+Weight+Height, data=NHANES)
```
Comment (roninsightrx) [inline R/calc_creat.R:12]: Nitpick, for proper rendering in help:
```suggestion
#' data and takes into account sex, age, weight, and height. 
Equation: `lm(SCr~Age+Sex+Weight+Height, data=NHANES)`
```

**Review by roninsightrx — APPROVED** [review]
if you move the sub-function out of main function it looks good to me.

**Comment by jasmineirx** [comment]
One last step, sorry that I just caught it. You need to run the command `devtools:document()` in order to render the new documentation you added. This will create a 'md' file, which you will need to commit and push up!

**Comment by JordanBrooks33** [comment]
> One last step, sorry that I just caught it. You need to run the command `devtools:document()` in order to render the new documentation you added. This will create a 'md' file, which you will need to commit and push up!

Nice catch! I have run the code and committed/pushed up the new documentation. I have not done this before, so just to double check -> in running the line above it updated the Description doc and created 2 new .rd docs in the man/ directory. Is this what was expected?

**Review by jasmineirx — APPROVED** [review]
lgtm! nice work :)

**Pull request merged**